pkgs/desktops/kde-5/applications/akonadi.nix

```diff
@@ -0,0 +1,44 @@
+{ boost
+, ecm
+, kdeApp
+, kdesignerplugin
+, kitemmodels
+, lib
+, makeQtWrapper
+, mariadb
+, postgresql
+, shared_mime_info
+}:
+
+kdeApp {
+  name = "akonadi";
+  meta = {
+    description = "KDE PIM Storage Service";
+    license = with lib.licenses; [ gpl2 ];
+    homepage = http://pim.kde.org/akonadi;
+    maintainers = with lib.maintainers; [ vandenoever ];
+  };
+  nativeBuildInputs = [
+    ecm
+    makeQtWrapper
+  ];
+  buildInputs = [
+    boost
+    kdesignerplugin
+    kitemmodels
+  ];
+  propagatedBuildInputs = [
+    mariadb
+    postgresql
+  ];
+
+  postInstall = ''
+    wrapQtProgram "$out/bin/akonadi_agent_launcher"
```

Comment:
`paths` is for extra runtime dependencies, to break circular dependency chains. For example, `dolphin-plugins` depends on `dolphin`, but `dolphin` needs to be wrapped to find the plugins. So, we make `dolphin-plugins` depend on `dolphin.unwrapped` and only the `dolphin` wrapper depends on `dolphin-plugins`.

For packages that depend on `akonadi` at compile time (i.e. they need headers, CMake files, etc.) put `akonadi.unwrapped` in their `buildInputs` instead.